### ISSUE TITLE
Two browsers opened when using TestNG's timeOut

### ISSUE THREAD

**Issue by ken-p** [codeborne/selenide#171]
This is an interesting one.  Have a base class, that handles creating and setting a webdriver:
```java
public class SomeBaseTestNGTest {
    @BeforeClass
    public void before() {
        setWebDriver(new FirefoxDriver()); //using com.codeborne.selenide.WebDriverRunner
    }
}
```
And in a test class, have a test that uses TestNG's timeout:
```java
public class SelenideAndTestNGTimeouts extends SomeBaseTestNGTest {
    @Test(timeOut = 5000)
    public void thisWillLaunchTwoBrowsers() {
        open("http://google.com");
        close();
    }
}
```

Right click and run the TestNG test, and it will open two browsers.  I'm guessing using the TestNG timeOut is spawning the test class into another thread, Selenide sees that thread not having a WebDriver associated with it, and so it creates one.

I think there's another ticket to have a feature to disable the automatic spawning of a browser.  I'm managing all of the browser lifecycles via base classes and factory helpers (and then setting via setWebDriver()), so that would be one way to work around this.

**Comment by asolntsev** [comment]
I guess you are right, TestNG runs a separate thread when `timeOut = 5000` is declared. We can make `WebDriverRunner` configurable, so that it always uses the same webdriver instance and does not create new ones. But you will not be able to run tests in parallel. Is it ok for you? Don't you want to run tests in parallel? Is this `timeOut` really useful for you?

**Comment by ken-p** [comment]
We have a mix of tests that run in parallel and run serially.  I've just started experimenting with TestNG's timeOut.  But I think for now actually using the Stopwatch class from guava will probably be a better fit for us as it's more granular and gives us more control.  So I don't think any action on this bug is necessary, at least from my point of view.  Thanks!

**Comment by asolntsev** [comment]
Ok. Thank you for knowledge exchange!

**Comment by asolntsev** [comment]
Ok, closing the issue.
